prime-dotnet-webapi/Services/EmailService.cs

```diff
@@ -246,6 +246,39 @@ public async Task SendEnrolleeRenewalEmails()
             }
         }
 
+
+        public async Task SendEnrolleeUnsignedToaReminderEmails()
+        {
+            var enrollees = await _context.Enrollees
+                .Include(e => e.EnrolmentStatuses)
+                .Select(e => new
```

Comment:
doing the `.Select` statement first and then the `.Where` afterwards causes extra SQL to be emitted by EF, since the `CurrentStatus` needs to be both SELECTed and WHEREd. Compare the generated SQL:
```sql
SELECT e1."FirstName", e1."LastName", e1."Email", (
    SELECT e."StatusDate"
    FROM "EnrolmentStatus" AS e
    WHERE (e1."Id" = e."EnrolleeId") AND (e."StatusCode" = 3)
    ORDER BY e."StatusDate" DESC, e."Id" DESC
    LIMIT 1) AS "ApprovedDate", (
    SELECT e0."StatusCode"
    FROM "EnrolmentStatus" AS e0
    WHERE e1."Id" = e0."EnrolleeId"
    ORDER BY e0."StatusDate" DESC, e0."Id" DESC
    LIMIT 1) AS "StatusCode"
FROM "Enrollee" AS e1
WHERE ((
    SELECT e2."StatusDate"
    FROM "EnrolmentStatus" AS e2
    WHERE (e1."Id" = e2."EnrolleeId") AND (e2."StatusCode" = 3)
    ORDER BY e2."StatusDate" DESC, e2."Id" DESC
    LIMIT 1) IS NOT NULL) AND ((
    SELECT e3."StatusCode"
    FROM "EnrolmentStatus" AS e3
    WHERE e1."Id" = e3."EnrolleeId"
    ORDER BY e3."StatusDate" DESC, e3."Id" DESC
    LIMIT 1) = 3)
```
to this SQL:
```sql
SELECT e0."FirstName", e0."LastName", e0."Email", (
    SELECT e."StatusDate"
    FROM "EnrolmentStatus" AS e
    WHERE (e0."Id" = e."EnrolleeId") AND (e."StatusCode" = 3)
    ORDER BY e."StatusDate" DESC, e."Id" DESC
    LIMIT 1) AS "ApprovedDate"
FROM "Enrollee" AS e0
WHERE (
    SELECT TRUE
    FROM "EnrolmentStatus" AS e1
    WHERE (e0."Id" = e1."EnrolleeId") AND (e1."StatusCode" = 3)
    ORDER BY e1."StatusDate" DESC, e1."Id" DESC
    LIMIT 1) AND ((
    SELECT e2."StatusCode"
    FROM "EnrolmentStatus" AS e2
    WHERE e0."Id" = e2."EnrolleeId"
    ORDER BY e2."StatusDate" DESC, e2."Id" DESC
    LIMIT 1) = 3)
```
generated when the `.Where` clause is moved up:
```C#
var enrollees = await _context.Enrollees
    .Where(e => e.ApprovedDate.HasValue
        && e.CurrentStatus.StatusCode == (int)StatusType.RequiresToa)
    .Select(e => new
    {
        e.FirstName,
        e.LastName,
        e.Email,
        e.ApprovedDate
    })
    .DecompileAsync()
    .ToListAsync();
```